### PR TITLE
Add support for the tests on national cloud.

### PR DESCRIPTION
**Pull request by pallavit**
Update ACR tests to work with national cloud(s). I have also added a manual weekly pipeline that runs it against USGov, China.

### CODE REVIEW

File: sdk/containerregistry/azure-containers-containerregistry/src/test/java/com/azure/containers/containerregistry/ContainerRepositoryAnonymousAccessTests.java [hunk 1]
```diff
@@ -4,6 +4,8 @@
 package com.azure.containers.containerregistry;
 
 import com.azure.core.http.HttpClient;
+import com.azure.core.test.implementation.ImplUtils;
```
Comment (alzimmermsft) [inline sdk/containerregistry/azure-containers-containerregistry/src/test/java/com/azure/containers/containerregistry/ContainerRepositoryAnonymousAccessTests.java:7]: nit: this shouldn't be accessed

File: sdk/containerregistry/tests.yml [hunk 2]
```diff
@@ -4,13 +4,14 @@ stages:
...
           groupId: com.azure
           safeName: azurecontainerscontainerregistry
       TimeoutInMinutes: 90
       EnvVars:
-        AZURE_CLIENT_ID: $(aad-azure-sdk-test-client-id)
-        AZURE_CLIENT_SECRET: $(aad-azure-sdk-test-client-secret)
-        AZURE_TENANT_ID: $(aad-azure-sdk-test-tenant-id)
+        AZURE_CLIENT_ID: $(CONTAINERREGISTRY_CLIENT_ID)
+        AZURE_CLIENT_SECRET: $(CONTAINERREGISTRY_CLIENT_SECRET)
+        AZURE_TENANT_ID: $(CONTAINERREGISTRY_TENANT_ID)
```
Comment (alzimmermsft) [inline sdk/containerregistry/tests.yml:16]: Are these values being outputted by the resource deployment script?

File: sdk/containerregistry/azure-containers-containerregistry/src/test/java/com/azure/containers/containerregistry/ContainerRegistryClientIntegrationTests.java [hunk 3]
```diff
@@ -176,19 +178,22 @@ public void convenienceProperties(HttpClient httpClient) {
...
-                .authenticationScope(AZURE_GOV_AUTHENTICATION_SCOPE)
-                .buildClient();
-            assertThrows(ClientAuthenticationException.class, () -> throwableRegistryClient.listRepositoryNames().stream().collect(Collectors.toList()));
+            if (getTestMode() != TestMode.PLAYBACK) {
+                // Now doing the same should fail with the separate registryClient;
+                ContainerRegistryClient throwableRegistryClient = getContainerRegistryBuilder(httpClient)
+                    .authenticationScope(AZURE_GOV_AUTHENTICATION_SCOPE)
+                    .buildClient();
+                assertThrows(ClientAuthenticationException.class, () -> throwableRegistryClient.listRepositoryNames().stream().collect(Collectors.toList()));
+            }
```
Comment (alzimmermsft) [inline sdk/containerregistry/azure-containers-containerregistry/src/test/java/com/azure/containers/containerregistry/ContainerRegistryClientIntegrationTests.java:196]: Should this be made into a separate test that has an assumption of `TestMode != PLAYBACK`

File: sdk/containerregistry/azure-containers-containerregistry/src/test/java/com/azure/containers/containerregistry/ContainerRepositoryAnonymousAccessTests.java [hunk 4]
```diff
@@ -21,14 +23,16 @@
...
+        TestUtils.importImageAsync(ImplUtils.getTestMode(), ANONYMOUS_REGISTRY_NAME, HELLO_WORLD_REPOSITORY_NAME, Arrays.asList("latest", "v1", "v2", "v3", "v4"), ANONYMOUS_REGISTRY_ENDPOINT).block();
     }
 
     @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
     @MethodSource("getHttpClients")
     public void listAnonymousRepositories(HttpClient httpClient) {
-        ContainerRegistryClient client = getContainerRegistryBuilder(httpClient, null, ANONYMOUS_REGISTRY_ENDPOINT).buildClient();
-        List<String> repositories = client.listRepositoryNames().stream().collect(Collectors.toList());
-        assertTrue(repositories.stream().anyMatch(HELLO_WORLD_REPOSITORY_NAME::equals));
+        if (TestUtils.getAuthority(ANONYMOUS_REGISTRY_ENDPOINT) == AzureAuthorityHosts.AZURE_PUBLIC_CLOUD) {
```
Comment (alzimmermsft) [inline sdk/containerregistry/azure-containers-containerregistry/src/test/java/com/azure/containers/containerregistry/ContainerRepositoryAnonymousAccessTests.java:32]: This should be made into an assumption

Comment (pallavit) [reply sdk/containerregistry/azure-containers-containerregistry/src/test/java/com/azure/containers/containerregistry/ContainerRepositoryAnonymousAccessTests.java:32]: [TIL] Thanks!

File: sdk/containerregistry/azure-containers-containerregistry/src/test/java/com/azure/containers/containerregistry/ContainerRepositoryAnonymousAccessTests.java (hunk 1 above)
Comment (pallavit) [reply sdk/containerregistry/azure-containers-containerregistry/src/test/java/com/azure/containers/containerregistry/ContainerRepositoryAnonymousAccessTests.java:7]: Yup looks like a bad merge - cleaning it.

File: sdk/containerregistry/tests.yml (hunk 2 above)
Comment (pallavit) [reply sdk/containerregistry/tests.yml:16]: Yes.

**Review by alzimmermsft — APPROVED** [review]

**Pull request merged**